### PR TITLE
feat(bridges): integrate PostgreSQL into bridges

### PR DESCRIPTION
**Pull request by lafirest**
support both simple and batch query

### CODE REVIEW

File: lib-ee/emqx_ee_bridge/src/emqx_ee_bridge_pgsql.erl [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        local_topic => <<"local/topic/#">>,
+        resource_opts => #{
+            worker_pool_size => 8,
+            health_check_interval => ?HEALTHCHECK_INTERVAL_RAW,
+            auto_restart_interval => ?AUTO_RESTART_INTERVAL_RAW,
+            enable_batch => true,
+            batch_size => ?DEFAULT_BATCH_SIZE,
+            batch_time => ?DEFAULT_BATCH_TIME,
+            query_mode => async,
+            enable_queue => false,
```
Comment (thalesmg) [inline lib-ee/emqx_ee_bridge/src/emqx_ee_bridge_pgsql.erl:65]: Some of these options were removed in a [recently merged PR](https://github.com/emqx/emqx/pull/9642/).  Please rebase with the current master to pick up the changes.

```suggestion
            batch_size => ?DEFAULT_BATCH_SIZE,
            batch_time => ?DEFAULT_BATCH_TIME,
            query_mode => async,
```

File: lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_pgsql_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+                    {pgsql_config, PGConf},
+                    {pgsql_bridge_type, BridgeType},
+                    {pgsql_name, Name},
+                    {proxy_host, ProxyHost},
+                    {proxy_port, ProxyPort}
+                    | Config0
+                ],
+            Config;
+        false ->
+            {skip, no_pgsql}
```
Comment (thalesmg) [inline lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_pgsql_SUITE.erl:152]: Taking from this [PR](https://github.com/emqx/emqx/pull/9653/files):

```suggestion
            case os:getenv("IS_CI") of
                "yes" ->
                    throw(no_pgsql);
                _ ->
                    {skip, no_pgsql}
            end
```

File: lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_pgsql_SUITE.erl [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+        io_lib:format(
+            "bridges.~s.~s {\n"
+            "  enable = true\n"
+            "  server = ~p\n"
+            "  database = ~p\n"
+            "  username = ~p\n"
+            "  password = ~p\n"
+            "  sql = ~p\n"
+            "  resource_opts = {\n"
+            "    enable_batch = ~p\n"
```
Comment (thalesmg) [inline lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_pgsql_SUITE.erl:172]: Now, `batch_size > 1 => enable_batch`:

```suggestion
            "    batch_size = ~b\n"
```

File: lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_pgsql_SUITE.erl [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+    BridgeType = ?config(pgsql_bridge_type, Config),
+    Name = ?config(pgsql_name, Config),
+    emqx_bridge:remove(BridgeType, Name).
+
+create_bridge_http(Params) ->
+    Path = emqx_mgmt_api_test_util:api_path(["bridges"]),
+    AuthHeader = emqx_mgmt_api_test_util:auth_header_(),
+    case emqx_mgmt_api_test_util:request_api(post, Path, "", AuthHeader, Params) of
+        {ok, Res} -> {ok, Res};
+        %%{ok, emqx_json:decode(Res, [return_maps])};
```
Comment (thalesmg) [inline lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_pgsql_SUITE.erl:216]:
```suggestion
        {ok, Res} -> {ok, emqx_json:decode(Res, [return_maps])};
```
?

File: apps/emqx_connector/src/emqx_connector_pgsql.erl [hunk 5]
```diff
@@ -116,37 +140,145 @@ on_stop(InstId, #{poolname := PoolName}) ->
...
+        [{Key, _} = Request | _] ->
+            BinKey = to_bin(Key),
+            case maps:get(BinKey, Tokens, undefined) of
+                undefined ->
+                    Log = #{
+                        connector => InstId,
+                        first_request => Request,
+                        state => State,
+                        msg => "batch prepare not implemented"
+                    },
```
Comment (thalesmg) [inline apps/emqx_connector/src/emqx_connector_pgsql.erl:186]: Nit: maybe we could log `BinKey` as well, so one can have more context if this fails?

Comment (lafirest) [reply apps/emqx_connector/src/emqx_connector_pgsql.erl:186]: It is already in the `Request`

**Review by qzhuyan — APPROVED** [review]

**Pull request merged**